hivemq-edge/src/main/java/com/hivemq/configuration/entity/adapter/NorthboundMappingEntity.java

```diff
@@ -160,15 +120,82 @@ public void validate(final @NotNull List<ValidationEvent> validationEvents) {
             case EXACTLY_ONCE -> 2;
         };
 
-        return new NorthboundMappingEntity(
-                northboundMapping.getTagName(),
+        return new NorthboundMappingEntity(northboundMapping.getTagName(),
                 northboundMapping.getTopic(),
                 qos,
                 MessageHandlingOptions.MQTTMessagePerTag,
                 northboundMapping.getIncludeTagNames(),
                 northboundMapping.getIncludeTimestamp(),
                 mqttUserPropertyEntities,
-                northboundMapping.getMessageExpiryInterval() != null ? northboundMapping.getMessageExpiryInterval() : Long.MAX_VALUE);
+                northboundMapping.getMessageExpiryInterval() != null ?
+                        northboundMapping.getMessageExpiryInterval() :
+                        Long.MAX_VALUE);
+    }
+
+    public static @NotNull NorthboundMappingEntity fromPollingContext(PollingContext ctx) {
+        final List<MqttUserPropertyEntity> mqttUserProperties = ctx.getUserProperties()
+                .stream()
+                .map(mqttUserPropertyEntity -> new MqttUserPropertyEntity(mqttUserPropertyEntity.getName(),
+                        mqttUserPropertyEntity.getValue()))
+                .collect(Collectors.toList());
```

Comment:
`.collect(Collectors.toList());`. -> `.toList();` 

more readable

Reply:
Aha, I was just formatting the code. There are too many of these.